gcp_query_tools/connect_analytics_helper_functions.R

```diff
@@ -245,52 +246,48 @@ add_novel_vars_to_csv <- function(vars, in_file, out_file = NULL) {
-  # Returns:
-  #   unique_responses: A character array of the unique values or responses 
-  #                     returned by the variable across all rows of the table.
-  # TODO: Take an array of variables as an input and write one query for all 
-  #       of them. There is a 10MB minimum charge, so combining running one 
-  #       query for all of the variables would be more cost efficient.
-  
-  # Query data for this variable
-  tab_path <- paste0("`", project, ".", table, "`") # Format table path for SQL
-  bq_query <- paste("SELECT", var_name, "AS var FROM", tab_path,  
-                    "WHERE",  var_name, "IS NOT NULL", sep=" ")
-  bq_table <- bq_project_query(project, bq_query) # 
-  df_var   <- bq_table_download(bq_table, bigint = "integer64")
-  
-  # Build list of unique responses for this variable
-  unique_responses <- c()
-  for (i in 1:nrow(df_var)){
-    responses           <- df_var$var[[i]]
-    responses_available <- length(responses) > 0
-    if (responses_available){
-      for (j in 1:length(responses)){
-        response          <- responses[j]
-        response_is_novel <- !(response %in% unique_responses)
-        if (response_is_novel){
-          unique_responses <- append(unique_responses, response)
-        }
-      }
+
+get_unique_values <- function(project, table, array_vars) {
+  
+  # Initialize a string to store the SQL query
+  sql_query <- ""
+  
+  # Loop through each array variable to generate the SQL query
+  for (array_var in array_vars) {
+    
+    # Add to the SQL query a SELECT DISTINCT statement to get unique values for the array variable
+    sql_query <- glue("{sql_query}
+      SELECT DISTINCT
+        '{array_var}' AS variable,  # Label the array variable
+        array_element               # Select distinct elements in the array
+      FROM
+        `{project}.{table}`,        # Specify the BigQuery table
+        UNNEST({array_var}) AS array_element      # UNNEST the array to transform it into rows
+    ")
+    
+    # If the current variable is not the last in the list, add UNION ALL to combine results
+    if (array_var != tail(array_vars, 1)) {
+      sql_query <- glue("{sql_query}
+        UNION ALL
+      ")
     }
   }
-  return(unique_responses)
+  
+  #Execute the SQL query on BigQuery and download the results
+  bq_table <- bq_project_query(project, query = sql_query)
+  df_var <- bq_table_download(bq_table, bigint = "integer64")
+  
+  #Convert the results into a list where each element corresponds to a variable
+  results_list <- split(as.numeric(df_var$array_element), df_var$variable)
+  
+  # Return the list of unique values for each variable
+  return(results_list)
 }
 
-filter_vars_from_schema <- function(project, table, schema, out_csv, out_json, 
+
+filter_vars_from_schema <- function(project, table, schema, out_csv, out_json, GUV_out_json,
                                     output_file_type = "json"
```

Comment:
I have a few suggestions here:
- [ ] rename GUV_out_json to out_json and remove the redundant argument
- [ ] remove the output_file_type argument now that we no longer need optional file types of "json" or "js"